src/app/clusters/general-diagnostics-server/GeneralDiagnosticsCluster.cpp

```diff
@@ -274,6 +275,26 @@ DataModel::ActionReturnStatus GeneralDiagnosticsCluster::ReadAttribute(const Dat
     case GeneralDiagnostics::Attributes::TestEventTriggersEnabled::Id: {
         bool isTestEventTriggersEnabled = IsTestEventTriggerEnabled();
         return encoder.Encode(isTestEventTriggersEnabled);
+    }
+    case GeneralDiagnostics::Attributes::DeviceLoadStatus::Id: {
+        static_assert(CHIP_IM_MAX_NUM_SUBSCRIPTIONS <= UINT16_MAX,
+                      "The maximum number of IM subscriptions is larger than expected (should fit within a 16 bit unsigned int)");
+        VerifyOrReturnError(mClusterContext.interactionModelEngine != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrReturnError(mClusterContext.sessionManager != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrReturnError(mClusterContext.reportScheduler != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+
+        const SessionManager::MessageStats messageStatistics = mClusterContext.sessionManager->GetMessageStats();
+
+        GeneralDiagnostics::Structs::DeviceLoadStruct::Type load = {
+            .currentSubscriptions = static_cast<uint16_t>(
+                mClusterContext.interactionModelEngine->GetNumActiveReadHandlers(ReadHandler::InteractionType::Subscribe)),
+            .currentSubscriptionsForFabric = static_cast<uint16_t>(mClusterContext.interactionModelEngine->GetNumActiveReadHandlers(
+                ReadHandler::InteractionType::Subscribe, encoder.AccessingFabricIndex())),
+            .totalSubscriptionsEstablished = mClusterContext.reportScheduler->GetTotalSubscriptionsEstablished(),
```

Comment:
Why does this live on the report scheduler and not just on the InteractionModelEngine?  Is the issue that tracking this in the IME would be hard because the IME knows about subscription establishment _attempts_ but does not know whether they succeed or fail?

But even then, it seems like the "public" API should be the IME; the fact that it keeps track of it via the report scheduler is implementation detail.  And then we would not need to inject a report scheduler here separately....